tox.ini

```diff
@@ -23,134 +23,95 @@ passenv =
+    poetry install --with integration
+    poetry export -f requirements.txt -o requirements.txt
+    poetry run pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0 -m db_admin_relation_tests
 
 [testenv:ha-self-healing-integration]
 description = Run high availability self healing integration tests
-deps =
-    juju==2.9.11 # juju 3.0 has issues with retrieving action results and deploying charms
-    kubernetes
-    pytest
-    pytest-operator
-    psycopg2-binary
-    -r{toxinidir}/requirements.txt
 commands =
-    pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0 -m ha_self_healing_tests
+    poetry install --with integration
+    poetry export -f requirements.txt -o requirements.txt
+    poetry run pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0 -m ha_self_healing_tests
 
 [testenv:password-rotation-integration]
 description = Run password rotation integration tests
-deps =
-    juju==2.9.11 # juju 3.0 has issues with retrieving action results and deploying charms
-    pytest
-    pytest-operator
-    psycopg2-binary
-    -r{toxinidir}/requirements.txt
 commands =
-    pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0 -m password_rotation_tests
+    poetry install --with integration
+    poetry export -f requirements.txt -o requirements.txt
+    poetry run pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0 -m password_rotation_tests
 
 [testenv:tls-integration]
 description = Run TLS integration tests
-deps =
-    juju==2.9.11 # juju 3.0 has issues with retrieving action results and deploying charms
-    pytest
-    pytest-operator
-    psycopg2-binary
-    -r{toxinidir}/requirements.txt
 commands =
-    pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0 -m tls_tests
+    poetry install --with integration
+    poetry export -f requirements.txt -o requirements.txt
+    poetry run pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0 -m tls_tests
 
 [testenv:integration]
 description = Run all integration tests
-deps =
-    juju==2.9.11 # juju 3.0 has issues with retrieving action results and deploying charms
-    pytest
-    pytest-operator
-    psycopg2-binary
-    -r{toxinidir}/requirements.txt
 commands =
-    pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0
+    poetry install --with integration
+    poetry export -f requirements.txt -o requirements.txt
+    poetry run pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs} --durations=0
+
+[testenv:poetry-lock]
+description = Install, lock and export poetry dependencies
+commands =
+    poetry lock
+    poetry export -f requirements.txt -o requirements.txt
```

Comment:
Auxiliary command so that we can update the lock file from `tox` when changing `pyproject.toml`. We can do this with an external, system installed poetry as well.